# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/crashtests/1517028.html
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 editor/libeditor/crashtests/1517028.html
--- a/editor/libeditor/crashtests/1517028.html	Thu Sep 22 06:24:38 2022 +0000
+++ b/editor/libeditor/crashtests/1517028.html	Thu Sep 22 06:27:37 2022 +0000
@@ -2,16 +2,21 @@
 abc
 <head>
 <script>
-  function start() {
-    document.execCommand('justifyleft', false, ) 
-    document.designMode = 'on' 
-    document.execCommand('insertparagraph', false, ) 
-  }
-  document.addEventListener('DOMContentLoaded', start)
-</script>  
+document.addEventListener('DOMContentLoaded', () => {
+  // For emulating the traditional behavior, collapse Selection to end of the
+  // <em> which is the deepest last child of the <body> (at the comment node).
+  getSelection().collapse(
+    document.querySelector("em[contenteditable]"),
+    document.querySelector("em[contenteditable]").childNodes.length
+  );
+  document.execCommand("justifyLeft");
+  document.designMode = 'on';
+  document.execCommand("insertParagraph");
+});
+</script>
 </head>
 <h4>
-<em contenteditable="">
+<em contenteditable>
 <big>
 <button autofocus formnovalidate formtarget="">
 </button>